### PR TITLE
Add developer info for creating a new JLL release

### PR DESCRIPTION
**Pull request by sloede**

### CODE REVIEW

File: development.md [hunk 1]
```diff
@@ -35,3 +35,37 @@ To create a new release for smesh, perform the following steps:
...
+     clicking on the short commit hash, then copying it from the browser URL.
+   * Change the existing commit hash in `build_tarballs.jl` to the commit hash you just
+     obtained (e.g.,
+     [here](https://github.com/JuliaPackaging/Yggdrasil/blob/414237372f5bac40fc3cd8045727def18388a1d7/S/smesh/build_tarballs.jl#L11)
+     the hash was `db69bf884d10c52577aee6795202136cfcf77178`)
+3) Commit and push the changes to your fork.
+4) Create a pull request from your fork's branch to Yggdrasil's `master` branch and name it
+   appropriately (e.g., `[smesh] update to version v0.2.4`)
+5) Wait for all tests in Yggdrasil to pass. If they don't, find the error and fix it.
+6) Wait for the Yggdrasil PR to be merged to `master`.. After an appropriate waiting period
```
Comment (JoshuaLampert) [inline development.md]:
```suggestion
6) Wait for the Yggdrasil PR to be merged to `master`. After an appropriate waiting period
```

File: development.md [hunk 2]
```diff
@@ -35,3 +35,37 @@ To create a new release for smesh, perform the following steps:
    `-dev` suffix added. For example, if you just released `v0.3.0`, the new development
    version should be `v0.3.1-dev`. If you just released `v0.2.4`, the new development
    version should be `v0.2.5-dev`.
+
+Among other uses, smesh is the backend for the Julia package
+[Smesh.jl](https://github.com/trixi-framework/Smesh.jl). For this purpose, it is
+automatically built and distributed as a Julia JLL package. Therefore, after each release of
+mesh, you should also update the corresponding build recipe in the
```
Comment (JoshuaLampert) [inline development.md]:
```suggestion
smesh, you should also update the corresponding build recipe in the
```

**Review by JoshuaLampert — COMMENTED** [review]
I just came across smesh and I'm excited to see it in action!

**Comment by sloede** [comment]
> I just came across smesh and I'm excited to see it in action!

Thanks for the review!

**Pull request merged**